tai_unpack.c

```diff
@@ -0,0 +1,18 @@
+/* Public domain. */
+
+#include "tai.h"
+
+void tai_unpack(const char *s,struct tai *t)
+{
+  uint64 x;
+
+  x = (unsigned char) s[0];
```

Comment:
This begs for a for loop.

Reply:
In my testing of a looped unpack function against an unrolled unpack function, the for-loop version was faster than this implementation about 11% of the time. This function is written the way it is for performance reasons. (I imagine rewriting it as a loop would cause it to get unrolled as a compiler optimization anyway; if so maybe the compiler could make it faster.)

Reply:
Compilers are clever beasts these days, they could even use SIMD instructions or similar things if they think it's useful.